### PR TITLE
Improve progress bar documentation

### PR DESCRIPTION
**Pull request by ExcaliburZero**
Improve the documentation for the ProgressBar type and its related functions.

### CODE REVIEW

File: Console/Display.hs [hunk 1]
```diff
@@ -184,7 +207,18 @@ progressStart pbar = do
...
+-- progress bar update function that is passed into `progress`.
+--
+-- @
+-- progressFunction :: ProgressBar -> IO String
+-- progressFunction bar = do
+--   threadDelay 100000
+--   progressTick bar
+--   threadDelay 200000
+--   progressTick bar
+--   return "Completed!"
```
Comment (NicolasDP) [inline Console/Display.hs:228]: you could have also referenced the example above I guess.
Can you also document what happens when the progression has reached `pgMax` ?

**Comment by NicolasDP** [comment]
looking good to me. What do you think @vincenthz  ?

File: Console/Display.hs (hunk 1 above)
Comment (ExcaliburZero) [inline Console/Display.hs:228]: Yep, noting what happens when the progression reaches the max would be a good idea. I'll also document what happens in cases where the max is exceeded, or the update function completes before the max is reached.

**Comment by ExcaliburZero** [comment]
I added some more information on how the progress bar fills, and also how it acts in cases in which more ticks are given than the specified max, and in the case that the update function completes before the bar is filled.

**Comment by vincenthz** [comment]
Thanks guys, nice documentation improvement.

**Pull request merged**